### ISSUE TITLE
question on architecture

### ISSUE THREAD

**Issue by ngfuong** [sauradip/STALE#11]
Thank you for your great paper.
I have some questions about network architecture and its code implementation of Representation Masking.

1. In paper, it's said that a MLP is used to optimize query per location. In class `STALE` I can spot a denifition of `self.mask_MLP` but cannot find where it is used. Can you help me?
2. The inputs to the transformer decoder are the snippet embedddings `F_vis` and mask queries `N_z`. Isn't this `N_z` similar to learnable queries in MaskFormer and should correspond to 5/20/100/etc? Why is it set to 1 in the object construction of `self.masktrans = TransformerPredictor=(..., num_queries=1,...)`?

**Comment by ed-fish** [comment]
mask NLP is defined on line 90 of stale_model.py

```py
 self.mask_MLP = nn.Sequential(
            nn.Conv1d(5,1,1),
            nn.Sigmoid()
        )
```

**Comment by sauradip** [comment]
Hi because i was just testing out on a low end GPU so due to memory constraints i removed the MLP and set queries to 1 so that it becomes 1-1 , since we have many-to-one we need a MLP and number of queries > 1. Hope that helps.
